docs/source/howto/codes.rst

```diff
@@ -3,173 +3,199 @@
-You can check the Topics section about :ref:`defining processes <topics:processes:usage:defining>` if you want more information about setting up your `inputs` and `outputs` (covering validation, dynamic number of inputs, etc.).
+
+
+Note that there is no ``return`` statement in ``define``: the ``define`` method directly modifies the |spec| object it receives.
+For more details on setting up your `inputs` and `outputs` (covering validation, dynamic number of inputs, etc.) see the :ref:`Defining Processes <topics:processes:usage:defining>` topic.
 
 Preparing for submission
 ------------------------
 
-The :py:meth:`~aiida.engine.processes.calcjobs.calcjob.CalcJob.prepare_for_submission` method is used for two purposes.
-Firstly, it should create the input files, based on the input nodes passed to the calculation, in the format that the external code will expect.
-Secondly, the method should create and return a :py:class:`~aiida.common.datastructures.CalcInfo` instance that contains various instructions for the engine on how the code should be run.
-An example implementation, as shipped with ``aiida-core`` can be seen in the following snippet:
+The :py:meth:`~aiida.engine.processes.calcjobs.calcjob.CalcJob.prepare_for_submission` method has two jobs:
+Firstly, it creates the input files in the format the external code expects.
+And secondly, it returns a :py:class:`~aiida.common.datastructures.CalcInfo` object that contains instructions for the AiiDA engine on how the code should be run.
+For example:
 
 .. literalinclude:: ../../../aiida/calculations/arithmetic/add.py
     :language: python
     :pyobject: ArithmeticAddCalculation.prepare_for_submission
-    :dedent: 4
 
-Note that, unlike the |define| method, this one is implemented from scratch and so there is no super call.
-The external code that we are running with this |CalcJob| is ``bash`` and so to sum the input numbers ``x`` and ``y``, we should write a bash input file that performs the summation, for example ``echo $((x + y))``, where one of course has to replace ``x`` and ``y`` with the actual numbers.
-You can see how the snippet uses the ``folder`` argument, which is a |Folder| instance that represents a temporary folder on disk, to write the input file with the bash summation.
-It uses Python's string interpolation to replace the ``x`` and ``y`` placeholders with the actual values that were passed as input, ``self.inputs.x`` and ``self.inputs.y``, respectively.
+.. note:: Unlike the |define| method, the ``prepare_for_submission`` method is implemented from scratch and so there is no super call.
 
-.. note::
+We start by writing our simple bash script that sums the two numbers ``x`` and ``y``, using Python's string interpolation to replace the ``x`` and ``y`` placeholders with the actual values ``self.inputs.x`` and ``self.inputs.y`` that were passed by the user.
```

Comment:
I would prefer to avoid "tutorial-style" language in the how-tos. Any instructions that explicitly prompt the reader to follow along ("We start by...") should be reserved for the tutorial.

Reply:
So, replace by "Start by"?

Reply:
Anyhow, to some degree I think we do want the reader to follow along here, i.e. to write their first calcjob and parser using the example we provide.
I.e. this howto certainly does have some tutorial character.
However, we currently don't have a "tutorial" category in the docs, just an "introduction" category that contains the basic tutorial. 
I think we should probably leave this under "howto"

Reply:
This is from the wiki about the how-to sections:
> The how-to guides should provide concrete steps, but do not have to be as detailed as a tutorial and further are not required to demonstrate a specific recipe from start to finish, meaning that there is some expectation that users are generally familiar with AiiDA.

So, yes, they can have some tutorial-character, but I would prefer to not give the impression that users are *expected* to follow along as they read them. That's obviously a very nuanced distinction, so happy to discuss, and we can even consider to change our guidelines regarding this if there is consensus.